radix-engine/src/kernel/track.rs

```diff
@@ -595,17 +595,33 @@ impl<'s> FinalizingTrack<'s> {
         // Commit/rollback application state changes
         let mut to_persist = HashMap::new();
         let next_epoch = {
-            // FIXME: schema - update
-            let expected_schema_hash = hash("EpochChangeEvent");
+            // TODO: Simplify once ScryptoEvent trait is implemented
+            let expected_event_name = {
+                let (local_type_index, schema) = generate_full_schema_from_single_type::<
+                    EpochChangeEvent,
+                    ScryptoCustomTypeExtension,
+                >();
+                (*schema
+                    .resolve_type_metadata(local_type_index)
+                    .expect("Cant fail")
+                    .type_name)
+                    .to_owned()
+            };
             application_events
                 .iter()
                 .find(|(identifier, _)| match identifier {
                     EventTypeIdentifier(
-                        RENodeId::GlobalPackage(EPOCH_MANAGER_PACKAGE)
-                        | RENodeId::GlobalComponent(ComponentAddress::EpochManager(..)),
-                        NodeModuleId::SELF,
-                        schema_hash,
-                    ) if *schema_hash == expected_schema_hash => true,
+                        Emitter::Function(
+                            RENodeId::GlobalPackage(EPOCH_MANAGER_PACKAGE),
+                            NodeModuleId::SELF,
+                            ..,
+                        )
+                        | Emitter::Method(
+                            RENodeId::GlobalComponent(ComponentAddress::EpochManager(..)),
+                            NodeModuleId::SELF,
+                        ),
+                        event_name,
+                    ) if *event_name == expected_event_name => true,
```

Comment:
I feel like we should completely remove this logic from engine and let consumers find the events in `application_events` of the receipt.

Reply:
So, do you propose that the commit result should no longer have the `next_epoch` section? Instead, to know if they should advance over to the next epoch they look for this event in the receipt.

Reply:
That's what I would like, althought it will require some change in node integration.